libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/extensions/coroutines/MapboxNavigation.kt

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.navigation.dropin.extensions.coroutines
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.route.RouterCallback
+import com.mapbox.navigation.base.route.RouterFailure
+import com.mapbox.navigation.base.route.RouterOrigin
+import com.mapbox.navigation.core.MapboxNavigation
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+data class RoutesWithOrigin(
+    val routes: List<DirectionsRoute>,
+    val routerOrigin: RouterOrigin
+)
+
+class RequestRoutesError(
+    val reasons: List<RouterFailure>,
+    message: String
+) : Error(message)
```

Comment:
We would not make these public at any point right? 

Make them `internal`. Also thinking they should probably not exist 🤔 .

Reply:
Correct. They will be internal. I am not worried about the visibility just yet. We will address it later when we are closer to the release.

We need better, coroutine based interface for route fetching. The MapboxNavigation only provides archaic, callback based API that doesn't fit our needs.

`MapboxNavigation.requestRoutes()` returns multiple values via `onRoutesReady(routes: List<DirectionsRoute>, routerOrigin: RouterOrigin)` callback. The suspendible coroutine must return both success values as a result. So, to do so we can either return both values via kotlin [Pair](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/)\<List\<DirectionsRoute\>, RouterOrigin\> data class or custom value object that better fits our domain. 
What approach would you choose?

Reply:
`MapboxDropInRouteRequester` and `RouteFetchComponent` have already been merged in to handle route requesting/fetching for the drop-in UI.

Reply:
@cafesilencio I know it is. This PR proposes alternative approach.

Reply:
> I am not worried about the visibility just yet. We will address it later when we are closer to the release.

That's a valid way to think about it, but please default to least possible visibility, we can always go through the code and expose the things that should be public. It'd be very likely for us to miss some helpers, data classes, or other obscure structures and release them publicly unintentionally if we default to public and try to hide later.